### PR TITLE
Refactor large workflows into multiple jobs

### PR DESCRIPTION
**Pull request by Racer159**
## Description

This PR refactors our larger workflows to use multiple jobs to avoid accumulation on disk.

## Related Issue

Fixes #N/A

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Other (security config, docs update, etc)

## Checklist before merging

- [X] Test, docs, adr added or updated as needed
- [X] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: .github/actions/cleanup-files/action.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    - run: |
+        lsblk -f
+
+        sudo rm -rf zarf-sbom /tmp/zarf-* src/ui/node_modules
+        sudo env "PATH=$PATH" CI=true make delete-packages
+        sudo build/zarf tools clear-cache
+        sudo docker system prune --all --force
+
+        lsblk -f
+      shell: bash
```
Comment (Noxsios) [inline .github/actions/cleanup-files/action.yaml:16]: weird YAML highlighting

Comment (Racer159) [reply .github/actions/cleanup-files/action.yaml:16]: Yeah I am not sure why that is - it is fine in VS code on the actual file

**Comment by Noxsios** [comment]
We could also throw:

```
func (e2e *ZarfE2ETest) SetupDockerRegistry(t *testing.T, port int) *configfile.ConfigFile {
	// spin up a local registry
	registryImage := "registry:2.8.2"
	err := exec.CmdWithPrint("docker", "run", "-d", "--restart=always", "-p", fmt.Sprintf("%d:5000", port), "--name", fmt.Sprintf("registry-%d", port), registryImage)
	require.NoError(t, err)
...

// TeardownRegistry removes the local registry.
func (e2e *ZarfE2ETest) TeardownRegistry(t *testing.T, port int) {
	// remove the local registry
	err := exec.CmdWithPrint("docker", "rm", "-f", fmt.Sprintf("registry-%d", port))
	require.NoError(t, err)
}
```

into this PR as well if you wanted @Racer159

**Review by Noxsios — APPROVED** [review]
lgtm, we will have to see how it performs on release

**Pull request merged**